diagnostics-app/src/app/core/services/events.service.ts

```diff
@@ -74,30 +79,44 @@ export class EventsService {
     });
   };
 
-  constructor() {
-    this.extensionId = environment.extensionId;
-
-    try {
-      //@ts-ignore
-      this.port = window.chrome.runtime.connect(
-        this.extensionId,
-        { name: PortName.EVENTS_PORT }
-      );
-
-      this.port.onMessage.addListener((msg: EventMessage) => {
-        if (this.subjects.has(msg.type)) {
-          this.subjects.get(msg.type)!.next(msg.info);
-        } else {
-          console.error(ResponseErrorInfoMessage.MissingEventsTypeSubject);
+  Init(): Promise<void>{
+    return new Promise<void>((resolve, reject) => {
+      try {
+        //@ts-ignore
+        this.port = window.chrome.runtime.connect(
+          this.extensionId,
+          { name: PortName.EVENTS_PORT }
+        );
+        this.port.onMessage.addListener((msg: EventMessage) => {
+          if (this.subjects.has(msg.type)) {
+            this.subjects.get(msg.type)!.next(msg.info);
+          } else {
+            console.error(ResponseErrorInfoMessage.MissingEventsTypeSubject);
+          }
+        });
+        for (const category of VISIBLE_EVENT_CARDS) {
+          this.isEventSupported(category);
         }
-      });
+        resolve();
+      } catch (err) {
+        console.error(ResponseErrorInfoMessage.FailedEventsServiceConstructor);
+        reject();
+      }
+    })
+  }
 
-    } catch (err) {
-      console.error(ResponseErrorInfoMessage.FailedEventsServiceConstructor);
-    }
+  constructor() {
+    this.extensionId = environment.extensionId;
+    this.supportabilityCache = new Map<EventCategory, Promise<EventsResponse>>;
```

Comment:
We can just initialize these in line 47:
```
private readonly extensionId = environment.extensionId;
private supportabilityCache = new Map<EventCategory, Promise<EventsResponse>>;
```

Reply:
Done.